### PR TITLE
shantell-sans: init at 1.011

### PR DESCRIPTION
**Pull request by baraahomsi**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/data/fonts/shantell-sans/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    install -D -t $out/share/fonts/truetype/ "$(find $src -type f -name '*.ttf')"
+    mv $out/share/fonts/truetype/"ShantellSans[BNCE,INFM,SPAC,ital,wght].ttf" $out/share/fonts/truetype/Shantell_Sans_Variable.ttf
+  '';
+
+  meta = with lib; {
+    homepage = "https://shantellsans.com/";
+    description = "A marker-style font built for creative expression, typographic play, and animation.";
+    license = licenses.ofl;
+    maintainers = [ maintainers.bodby ];
+    platforms = platforms.all;
```
Comment (Aleksanaa) [inline pkgs/data/fonts/shantell-sans/default.nix:28]:
```suggestion
  meta = {
    homepage = "https://shantellsans.com/";
    description = "A marker-style font built for creative expression, typographic play, and animation.";
    license = lib.licenses.ofl;
    maintainers = with lib.maintainers; [ bodby ];
    platforms = lib.platforms.all;
```

File: pkgs/data/fonts/shantell-sans/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  installPhase = ''
+    install -D -t $out/share/fonts/opentype/ $(find $src -type f -name '*.otf')
+
+    install -D -t $out/share/fonts/truetype/ "$(find $src -type f -name '*.ttf')"
+    mv $out/share/fonts/truetype/"ShantellSans[BNCE,INFM,SPAC,ital,wght].ttf" $out/share/fonts/truetype/Shantell_Sans_Variable.ttf
+  '';
+
+  meta = with lib; {
+    homepage = "https://shantellsans.com/";
+    description = "A marker-style font built for creative expression, typographic play, and animation.";
```
Comment (Aleksanaa) [inline pkgs/data/fonts/shantell-sans/default.nix:25]:
```suggestion
    description = "Marker-style font built for creative expression, typographic play, and animation";
```

File: pkgs/data/fonts/shantell-sans/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  fetchzip,
+  stdenvNoCC,
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "shantell-sans";
+  version = "1.011";
+
+  src = fetchzip {
+    url = "https://github.com/arrowtype/shantell-sans/releases/download/${version}/Shantell_Sans_${version}.zip";
```
Comment (Aleksanaa) [inline pkgs/data/fonts/shantell-sans/default.nix:12]:
```suggestion
stdenvNoCC.mkDerivation (finalAttrs: {
  pname = "shantell-sans";
  version = "1.011";

  src = fetchzip {
    url = "https://github.com/arrowtype/shantell-sans/releases/download/${finalAttrs.version}/Shantell_Sans_${finalAttrs.version}.zip";
```

File: pkgs/data/fonts/shantell-sans/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    hash = "sha256-xgE4BSl2A7yeVP5hWWUViBDoU8pZ8KkJJrsSfGRIjOk=";
+  };
+
+  installPhase = ''
+    install -D -t $out/share/fonts/opentype/ $(find $src -type f -name '*.otf')
+
+    install -D -t $out/share/fonts/truetype/ "$(find $src -type f -name '*.ttf')"
+    mv $out/share/fonts/truetype/"ShantellSans[BNCE,INFM,SPAC,ital,wght].ttf" $out/share/fonts/truetype/Shantell_Sans_Variable.ttf
+  '';
```
Comment (Aleksanaa) [inline pkgs/data/fonts/shantell-sans/default.nix:21]:
```suggestion
  installPhase = ''
    runHook preInstall
    install -D -t $out/share/fonts/opentype/ $(find $src -type f -name '*.otf')

    install -D -t $out/share/fonts/truetype/ "$(find $src -type f -name '*.ttf')"
    mv $out/share/fonts/truetype/"ShantellSans[BNCE,INFM,SPAC,ital,wght].ttf" $out/share/fonts/truetype/Shantell_Sans_Variable.ttf
    runHook postInstall
  '';
```

**Comment by bodby** [comment]
I would like to quickly ask some questions somewhat related to this, since I couldn't find any resources:

- How do you build (not test) default.nix packages like this with `nix-build`? I did it with `pkgs = import <nixpkgs> {};` but I feel like there should be a way to test it without modifying anything.
- Do I need to rename the .ttf to get rid of the brackets in its name?

Sorry if this is the wrong place to ask; I don't really know where else to go.

**Comment by Aleksanaa** [comment]
> How do you build (not test) default.nix packages like this with `nix-build`? I did it with `pkgs = import <nixpkgs> {};` but I feel like there should be a way to test it without modifying anything.

At project root of this repository, run `nix-build -A foo`

> Do I need to rename the .ttf to get rid of the brackets in its name?

Not sure, sorry. You should try locally to see if programs pick up this font correctly, I think

File: pkgs/data/fonts/shantell-sans/default.nix
Comment (Aleksanaa) [inline pkgs/data/fonts/shantell-sans/default.nix:1]: Move this to `pkgs/by-name/sh/shantell-sans/package.nix`

**Pull request closed without merging**